exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -1780,7 +1780,7 @@ func TestReconcileMachinePoolScaleToFromZero(t *testing.T) {
 				},
 			},
 		}
-		g.Expect(env.Create(ctx, node)).To(Succeed())
+		g.Expect(env.CreateAndWait(ctx, node)).To(Succeed())
```

Comment:
Nice catch!

Please also adjust the other occurences for `g.Expect(env.Create(ctx, node)).To(Succeed())` in this file.

While they probably don't lead to a flaky test right now, I think it's fine to also use CreateAndWait there

Reply:
@sbueringer Should we replace `env.Cleanup` with `env.CleanupAndWait` as well for that matter?

Reply:
@sbueringer Done